Upgrade EUI to v63.0.6 - updates to new EuiPageTemplate and deprecates old page components

cee-chen opened the pull request:
## Summary

`eui@62.2.4` ⏩ `eui@63.0.6`

----

<details><summary><code>63.0.1</code> ⏩ <code>63.0.6</code><br>Patch releases containing bugfixes for EuiPageTemplate and release fixes already present in the previous EUI PR</summary>

## [`63.0.6`](https://github.com/elastic/eui/tree/v63.0.6)

- Added a new `component` prop to `EuiPageSection`, allowing overriding of the default `section` tag ([#6192](https://github.com/elastic/eui/pull/6192))

**Deprecations**

- Added `@deprecated` flags to `EuiPageContent_Deprecated`, `EuiPageContentBody_Deprecated`, `EuiPageContentHeader_Deprecated`, `EuiPageContentHeaderSection_Deprecated`, `EuiPageSideBar_Deprecated` and `EuiPageTemplate_Deprecated`, which will provide helpful hints to IDEs that support jsdoc flags. Consumers will have until August 2023 to migrate from these deprecated components. ([#6194](https://github.com/elastic/eui/pull/6194))

## [`63.0.5`](https://github.com/elastic/eui/tree/v63.0.5)

**Bug fixes**

- Fixed missing `data-fixed-headers` property in some layout configurations using `EuiPageTemplate`. ([#6140](https://github.com/elastic/eui/pull/6140))
- Fixed an `EuiPageSidebar` bug where inline styles were not correctly updating ([#6191](https://github.com/elastic/eui/pull/6191))

## [`63.0.4`](https://github.com/elastic/eui/tree/v63.0.4)

**Bug fixes**

- Fixed the text size of `EuiDescriptionListTitle` when `EuiDescriptionList` is compressed ([#6160](https://github.com/elastic/eui/pull/6160))

## [`63.0.3`](https://github.com/elastic/eui/tree/v63.0.3)

**Bug fixes**

- Fixed `CollapsedItemActions` ref callback not accounting for `null` value ([#6145](https://github.com/elastic/eui/pull/6145))

## [`63.0.2`](https://github.com/elastic/eui/tree/v63.0.2)

**Bug fixes**

- Fixed `eui.d.ts` containing `@testing-library` type definitions ([#6142](https://github.com/elastic/eui/pull/6142))

## [`63.0.1`](https://github.com/elastic/eui/tree/v63.0.1)

**Bug fixes**

- Fixed server-side rendering and test-env errors caused by `useCurrentEuiBreakpoint` ([#6117](https://github.com/elastic/eui/pull/6117))
</details>

<details open><summary><code>63.0.0</code> - Main EuiPageTemplate release changelog</summary>

## [`63.0.0`](https://github.com/elastic/eui/tree/v63.0.0)

- Added new `EuiPageTemplate` namespaced component that uses context to pass through props ([#5768](https://github.com/elastic/eui/pull/5768))
- Added `EuiPageSection` component for easier section stacking ([#5768](https://github.com/elastic/eui/pull/5768))
- Added `EuiPageSidebar` component that is an upgrade to `EuiPageSideBar` ([#5768](https://github.com/elastic/eui/pull/5768))
- Extended `bottomBorder` prop to add `'extended'` on `EuiPageHeader` ([#5768](https://github.com/elastic/eui/pull/5768))
- Added `paddingSize` and `restrictWidth` directly on `EuiPageHeaderContent` ([#5768](https://github.com/elastic/eui/pull/5768))
- Added `data-fixed-headers` attribute to `<body>` with the count of fixed `EuiHeader` components ([#5768](https://github.com/elastic/eui/pull/5768))
- Extended `usePortal` prop on `EuiBottomBar` to accept `EuiPortalProps` ([#5768](https://github.com/elastic/eui/pull/5768))
- Increased `paddingSize` support by `EuiPage` for `xl` and `xs` sizes ([#5768](https://github.com/elastic/eui/pull/5768))
- Moved `restrictWidth` default `true` style to `style` attribute on `EuiPage` and `EuiPageBody` ([#5768](https://github.com/elastic/eui/pull/5768))
- Added optional `height` parameter to `euiYScroll()`, `useEuiYScroll`, `euiYScrollWithShadows()`, and `useEuiYScrollWithShadows` ([#5768](https://github.com/elastic/eui/pull/5768))
- Added `repositionOnScroll` directly to `EuiPopover` rendered by mobile version of `EuiHeaderLinks` ([#5768](https://github.com/elastic/eui/pull/5768))

**Bug fixes**

- Fixed missing render of `breadcrumbs` on `EuiPageHeader` when `alignItems = 'top'` ([#5768](https://github.com/elastic/eui/pull/5768))
- Fixed `logicalStyle()` to return the same value type as was passed in (instead of converting to string) ([#5768](https://github.com/elastic/eui/pull/5768))

**Deprecations**

- Deprecated the old `EuiPageTemplate` component and renamed to `EuiPageTemplate_Deprecated` ([#5768](https://github.com/elastic/eui/pull/5768))
- Deprecated `EuiPageContent` and `EuiPageContentBody` in favor of new `EuiPageSection` ([#5768](https://github.com/elastic/eui/pull/5768))
- Deprecated `EuiPageContent` and renamed to `EuiPageContent_Deprecated` ([#5768](https://github.com/elastic/eui/pull/5768))
- Deprecated `EuiPageContentBody` and renamed to `EuiPageContentBody_Deprecated` ([#5768](https://github.com/elastic/eui/pull/5768))
- Deprecated `EuiPageContentHeader` and renamed to `EuiPageContentHeader_Deprecated` ([#5768](https://github.com/elastic/eui/pull/5768))
- Deprecated `EuiPageContentHeaderSection` and renamed to `EuiPageContentHeaderSection_Deprecated` ([#5768](https://github.com/elastic/eui/pull/5768))
- Deprecated `EuiPageSideBar` and renamed to `EuiPageSideBar_Deprecated` ([#5768](https://github.com/elastic/eui/pull/5768))

**Breaking changes**

- Removed nested `EuiPageHeader` styles when a child of `EuiPageBody` ([#5768](https://github.com/elastic/eui/pull/5768))
- Changed default `paddingSize` of `EuiPage` from `m` to `none` ([#5768](https://github.com/elastic/eui/pull/5768))

**CSS-in-JS conversion**

- Converted `EuiPage` and `EuiPageBody` to Emotion ([#5768](https://github.com/elastic/eui/pull/5768))
- Converted `EuiPageHeader` and `EuiPageHeaderContent` to Emotion ([#5768](https://github.com/elastic/eui/pull/5768))
- Removed `euiPageRestrictWidth()` Sass mixin ([#5768](https://github.com/elastic/eui/pull/5768))
</details>

## EuiPageTemplate

This component has been replaced (in EUI) with a new **flexible** namespaced component that utilizes React context to pass through top level props and a new sectioning component that makes it easier to stack page contents including headers in any order.

👉  New component docs: https://elastic.github.io/eui/#/templates/page-template
👉  EUI PR with details: https://github.com/elastic/eui/pull/5768
👉  Guidelines: https://elastic.github.io/eui/#/templates/page-template/guidelines

In addition, the latest update deprecates a lot of (now defunct) page components like EuiPageContent and renames then with `_Deprecated`. This will hopefully alert owners that they should be switching to the latest template paradigm.

## Tasks performed

### ✅  Updated deprecated page component imports

Example:

```diff
import {
-  EuiPageContent,
-  EuiPageContentBody,
+  EuiPageContent_Deprecated as EuiPageContent,
+  EuiPageContentBody_Deprecated as EuiPageContentBody,
} from '@elastic/eui';
```

### ✅  Updated the `@kbn/shared-ux-page-kibana-template` version of KibanaPageTemplate

This is mostly done and even re-exports the namespaced components so that consumers in Kibana don't have to mis-match Kibana vs EUI prefixed components.

https://github.com/elastic/kibana/blob/2829dea45c970ef21188e399c4004ca40a02dca9/packages/kbn-shared-ux-components/src/page_template/page_template.tsx#L64-L70

### ✅  Updated the tutorial documentation page

New version found here: https://github.com/elastic/kibana/blob/2829dea45c970ef21188e399c4004ca40a02dca9/dev_docs/tutorials/kibana_page_template.mdx

### ✅  Fixed current usages of the Shared-UX KibanaPageTemplate

Only those consuming `KibanaPageTemplate` from `@kbn/shared-ux-page-kibana-template` have been fixed / updated to the new paradigm.

### ✅  Converted a few examples from old template to new

For changes affected by [9812251](https://github.com/elastic/kibana/pull/138546/commits/9812251c33e0674d6c607e6086d72489a8f28fa3), see Caroline's comment review in her original PR: https://github.com/elastic/kibana/pull/138546#pullrequestreview-1070369103

### ✅  Updated snapshots and FTR tests

The snapshots and tests caught some amount of bugs, but can't catch every possible UI bug or regression. Please take the time to manually QA/regression test your app.

## 🚨 Affected apps/teams 🚨

If your app/solution is in the below list list, it was affected by the new `EuiPageTemplate`/`KibanaPageTemplate` updates. **Please take the time to manually QA/UI regression test your app**.

- **Observability Solution**
- **Security Solution**
- **Enterprise Search Solution**
- Kibana Overview plugin
- Home plugin
- Management plugin
- Security plugin - API keys

## ⚠️ Teams that should eventually migrate from deprecated components ⚠️ 

Apps importing `KibanaPageTemplate` from `@kbn/kibana-react-plugin/public` are using the old/deprecated `EuiPageTemplate` and should convert to the `KibanaPageTemplate` from `@kbn/shared-ux-page-kibana-template` as soon as possible. This applies to the following apps:

- Canvas plugin
- ML plugin
- OSQuery plugin
- Spaces plugin - spaces selector

Teams that were directly using the now-deprecated `EuiPageTemplate`/`EuiPageContent` components (as noted by an `_Deprecated` import from EUI) **should generally convert to using KibanaPageTemplate**, or at minimum the new non-deprecated **EuiPage\*** equivalents. There are too many applications to list of these usages (222 files total), so we cannot call these teams out individually, but please consider doing a grep for `_Deprecated as` in your team codebase to check for them and [start planning to migrate over](https://elastic.github.io/eui/#/templates/page-template/guidelines#converting-from-legacy-layouts).

The deprecated page components will be removed by **August 2023**.

Commit:
Update all usages of EuiPageTemplate to EuiPageTemplate_Deprecated

Commit:
EuiPageContent_Deprecated as EuiPageContent

Commit:
EuiPageContentBody_Deprecated as EuiPageContentBody

Commit:
EuiPageContentHeader_Deprecated as EuiPageContentHeader

Commit:
EuiPageContentHeaderSection_Deprecated as EuiPageContentHeaderSection

Commit:
EuiPageSideBar_Deprecated as EuiPageSideBar

Commit:
EuiPageContent__Deprecated to EuiPageContent_Deprecated

Commit:
Fix rogue semi-colons

Commit:
WIP: NoDataConfigPage & NoDataPage converted to new template

- `withSolutionNav` not yet handled

Commit:
WIP: KibanaPageTemplateInner converted to use new template

- Pushes existing `pageHeader` prompts onto created EuiPageTemplate.PageHeader
- Uses `isEmptyState` to push `pageHeader` props to EuiPageTemplate.EmptyPrompt instead (if `children` are not supplied)

Commit:
WIP: `withSolutionNav` now renders the sidebar content properly

- Collapsing isn’t working (minWidth isn’t updating)

Commit:
Fixing stickiness of sidebar

Commit:
[Security] Fixed SecuritySolutionTemplateWrapper’s usage

- Moved `bottomBar` to EuiPageTemplate.BottomBar (now contained in just the page contents)
- Change EuiPanel children wrapper with EuiPageTemplate.Section

Commit:
[O11y] Wrap `children` with EuiPageTemplate.Section

Commit:
Fix getting_started usage

Commit:
WIP: Fixing types

Commit:
Removing `template` pass through

Commit:
Set EUI to 63.0.0

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint --no-cache --fix'

Commit:
More import fixes

Commit:
Sidebar component update

Commit:
Expand `KibanaPageTemplate` to all namespaced EUI counterparts

- Updated `docs/tutorials` mdx page
- Fixed SolutionNav prop types

Commit:
Updated the tutorial mdx page

Commit:
[Stack Management] Updated app layout to new template

- Some temporary props applied for BWC until all other pages can be converted
- Converted `API Keys` page’s layout (and especially prompt usags) to new paradigm

Commit:
Fix circular dep

cee-chen committed:
Fix new circular dependency

- copying and pasting types from KibanaPageTemplateProps, but ah well

cee-chen committed:
[Security Solution] Remove `template` prop - no longer a prop on Kibana/EuiPageTemplate

cee-chen committed:
[O11y] Allow customizing EuiPageTemplate.Section wrapper

- converts pageBodyProps
- fixes non-centered loading template

cee-chen committed:
[Enterprise Search] Update page templates

- fix layouts by auto-wrapping an EuiPageSection for padding, while adding a `customPageSections` prop for more custom sections/layouts

- re-center 404 errors

- update tests

cee-chen committed:
Update KibanaPageTemplate tests

cee-chen committed:
Update snapshots

cee-chen committed:
Fix FTR test with removed EUI classNames

cee-chen committed:
Fix FTR tests with changed kbn classNames

cee-chen committed:
Update failing dashboard snapshots

- drop shadow changed slightly in EUI

cee-chen committed:
Fix failing Security Cypress test

cee-chen committed:
[O11y] Fix Inventory page using deprecated CSS hooks

cee-chen committed:
[O11y][Uptime] Fix missing bottom bars

- Modifies ObservabilityPageTemplate to accept a `bottomBar` prop (a la the old EuiPageTemplate behavior)

NOTE: This opinionated page layout structure is starting to feel like it could be potentially limiting / have all the same pitfalls the previous EuiPageTemplate did. If so, consider something closer to the Enterprise Search page template conversion (`customPageSections`).

- Misc cleanup: Use `KibanaPageTemplate` over `EuiPageTemplate`

cee-chen committed:
[O11y] Fix route template typing

- Since theObservabilityPageTemplate is using the new Eui/KibanaPageTemplate, its child templates and types need to be updated accordingly

cee-chen committed:
Fix broken minWidth behavior

- was an EUI issue that required a patch release
+ update snapshots

cee-chen committed:
[Security Solution] Type fixes, restore empty state

- Fix empty state logic removed in a previous commit
- bogarts KibanaPageTemplate's `isEmptyState` prop instead of using `template="noData"`

- extend template wrappers to past ...rest to underlying Kibana/EuiPageTemplate

+ replace EuiPageTemplate with KibanaPageTemplate for consistency

cee-chen committed:
Fix failing synthetics selector

cee-chen committed:
Grab EUI v63.0.6

- for deprecation tags and section tag support

cee-chen committed:
Fix Kibana Overview plugin layout

- needs to use KibanaPageTemplate.Section to get padding back

- use `bottomBorder` prop over horizontal rules

- restore previous page color via panelled=false

cee-chen committed:
Convert Home plugin to new KibanaPageTemplate

- use KibanaPageTemplate.Section instead to preserve page width/paddings

- use `bottomBorder` instead of `EuiHorizontalRule`

- NOTE: This causes margins to decrease slightly from xxl to xl (largest padding available for EuiPageSection) - this can be restored by CSS overrides if desired

- update CSS to preserve previous looks, + convert to logical properties

constancecchen commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into test/page_template

constancecchen commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into test/page_template

constancecchen commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into test/page_template

elasticmachine commented:
Pinging @elastic/eui-design (EUI)

constancecchen commented:
👋 Hi everyone! This is a fairly large PR that primarily affects apps using `KibanaPageTemplate` from `@kbn/shared-ux-page-kibana-template` and deprecates several `EuiPage*` components.

## 🚨 Affected apps/teams 🚨

If your app/solution is in the below list list, it was affected by the new `EuiPageTemplate`/`KibanaPageTemplate` updates. **Please take the time to manually QA/UI regression test your app**.

- Observability Solution
- Security Solution
- Enterprise Search Solution
- Kibana Overview plugin
- Home plugin
- Management plugin
- Security plugin - API keys

## ⚠️ Teams that should eventually migrate from deprecated components ⚠️ 

Apps importing `KibanaPageTemplate` from `@kbn/kibana-react-plugin/public` are using the old/deprecated `EuiPageTemplate` and should convert to the `KibanaPageTemplate` from `@kbn/shared-ux-page-kibana-template` as soon as possible. This applies to the following apps:

- Canvas plugin
- ML plugin
- OSQuery plugin
- Spaces plugin - spaces selector

Teams that were directly using the now-deprecated `EuiPageTemplate`/`EuiPageContent` components (as noted by an `_Deprecated` import from EUI) **should generally convert to using KibanaPageTemplate**, or at minimum the new non-deprecated **EuiPage\*** equivalents. There are too many applications to list of these usages (222 files total), so we cannot call these teams out individually, but please consider doing a grep for `_Deprecated as` in your team codebase to check for them and [start planning to migrate over](https://elastic.github.io/eui/#/templates/page-template/guidelines#converting-from-legacy-layouts).

The deprecated page components will be removed by **August 2023**.

## CI failures/flakes

I have investigated and fixed all CI FTR/e2e failures except for osquery's cypress tests. I'm working with @tomsonpl to figure out if those failures are related to the EUI PR or flakes that exist in main (based on the changeset of this PR, we're leaning towards existing flakes).

If CI shows a failure, and only `Osquery Cypress Tests` is listed as failing, please ignore the failures and review as normal. Thank you!

jbudz reviewed (approved):
license_checker/config.ts

elasticmachine commented:
Pinging @elastic/apm-ui (Team:apm)

elasticmachine commented:
Pinging @elastic/fleet (Team:Fleet)

elasticmachine commented:
Pinging @elastic/uptime (Team:uptime)

constancecchen commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into test/page_template

tomsonpl commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into test/page_template

formgeist commented:
@constancecchen If we find regressions in our UI, do you want to try and fix those in this PR or create issues and follow-up after this has merged? I found some are pretty severe e.g. in the Logs view in Observability where the logs component is no longer readable - and the Observability navigation going away in the UX app.

jughosta reviewed (approved):
Data Discovery changes LGTM 👍 
Created an issue to address deprecated imports https://github.com/elastic/kibana/issues/139890

formgeist commented:
> @constancecchen If we find regressions in our UI, do you want to try and fix those in this PR or create issues and follow-up after this has merged? I found some are pretty severe e.g. in the Logs view in Observability where the logs component is no longer readable - and the Observability navigation going away in the UX app.

For now I've just created a temporary meta issue for collecting all regressions https://github.com/elastic/kibana/issues/139893

tomsonpl reviewed (approved):
@elastic/security-asset-management LGTM

jgowdyelastic reviewed (approved):
ML, Transforms and File data visualiser changes LGTM

cnasikas reviewed (approved):
ResponseOps changes LGTM

PhilippeOberti reviewed (approved):
LGTM for the Protections Experience team!

thomheymann reviewed (approved):
Security team changes LGTM! 

Thanks for updating all of these.

Bamieh reviewed (approved):
core changes lgtm

flash1293 reviewed (approved):
VisEditors changes LGTM, thanks for removing trailing semicolons!

paul-tavares reviewed (approved):
👍 on behalf of @elastic/security-onboarding-and-lifecycle-mgt  team.

I will create a tech debut for our team to (at some point) remove the use of the deprecated components and replace them with the new ones

constancecchen commented:
Thanks @formgeist for the super thorough QA! 🙏  At the risk of being annoying, would it be okay if we did a mix of "fix it now and fix it later post-PR"? I'm hoping to get this PR merged by early/mid next week, and will go through #139893 until then to try and help fix the listed issues you've found. Once we have all team approvals however, it would be great to merge as-is so we don't lose more time due to merge conflicts/flaky CI runs/etc. Hope that sounds good - I'll start looking into the logs layout issues now!

cee-chen committed:
[O11y] Fix non-centered empty/loading states

cee-chen committed:
[O11y] Restore subdued background on various empty state prompts

ThomThomson commented on src/plugins/dashboard/public/application/dashboard_app_no_data.tsx:
```diff
@@ -33,7 +33,7 @@ export const DashboardAppNoDataPage = ({
   return (
     <AnalyticsNoDataPageKibanaProvider {...analyticsServices}>
-      <AnalyticsNoDataPage onDataViewCreated={onDataViewCreated} />;
+      <AnalyticsNoDataPage onDataViewCreated={onDataViewCreated} />
```
Nice catch

constancecchen replied:
All credit to Caroline for finding those!

Re: snapshots - the dashboard snapshots did change in this PR because we tweaked panel drop shadow strength within EUI slightly. Unfortunately when I ran `--updateBaselines` on the FTR tests, my screenshots came out super large (2000px wide??) compared to the previous/CI screenshots. I'm not sure how to update the screenshots in a way that CI expects, really sorry 🙈 definitely happy to re-run the snapshot updates if there's any guidance you can give me in that direction!

ThomThomson replied:
This isn't the first time I've heard of someone having trouble updating the baselines for these tests! Let me look into it later, but for the time being upping the threshold is okay! Thanks for letting me know - I might need to work with the operations team to get this resolved.

ThomThomson reviewed (approved):
storyshot & snapshot changes LGTM. Also approving the screenshot testing threshold increase, but it is getting up there. Soon we should likely validate that the appearance is as expected, regenerate the screenshots, and lower the threshold.

cee-chen committed:
[O11y] Fix all instances of views that require a scrollable full-height child

+ restore comment for inventory view

cee-chen committed:
[O11y][ux] Fix broken sidebar

- The entire app was missing a wrapping EuiProvider, and as such breakpoint utils were not working, and the sidebar was missing

+ misc cleanup
  - remove unnecessary fragment
  - remove role="main" attr - now that EuiPageTemplate sets a `main` tag, they'll conflict
  - add isEmptyState to center loading component

cee-chen committed:
Merge remote-tracking branch 'upstream/main' into test/page_template

clintandrewhall commented on packages/shared-ux/page/kibana_template/impl/src/page_template_inner.tsx:
```diff
@@ -25,45 +25,60 @@ const getClasses = (template?: string, className?: string) => {
+  if (pageSideBar) {
+    sideBar = (
+      <EuiPageTemplate.Sidebar
+        // TODO: Get `offset` from Kibana Chrome Header and pass directly to:
```
@constancecchen Should we log an issue for this TODO?  I'm thinking so, and then reference it here.

clintandrewhall commented on packages/shared-ux/page/solution_nav/src/with_solution_nav.scss:
```diff
@@ -1,17 +1,8 @@
+// TODO: Can now be converted to Emotion
```
We should update https://github.com/elastic/kibana/issues/122594 and reference it here.

clintandrewhall commented on packages/shared-ux/page/solution_nav/src/with_solution_nav.tsx:
```diff
@@ -18,31 +18,30 @@ function getDisplayName(Component: ComponentType<any>) {
-type ComponentProps = TemplateProps & {
-  isEmptyState?: boolean;
-};
+// TODO: Would be nice to grab these from KibanaPageTemplate or vice-versa
```
If this is a follow-up for Shared UX team, we should file an issue and ref it here.  Perhaps a meta "follow up" issue to track them all, along with some details?

constancecchen replied:
Hey hey! For context, Caroline added these TODOs when she was first starting up the PR, so I'm much less familiar with their context/intention 🙈 Apologies for that - but yes, I'd be happy to file a follow-up issue from this PR for the Shared UX team. Thanks so much!

clintandrewhall reviewed (approved):
Thank you for your work on this!  Could we please file follow-up issues and reference them inline?

formgeist reviewed (approved):
Done a general pass and the found issues have been resolved. Minor issues will be tracked in separate issues.

cee-chen committed:
Merge branch 'main' into test/page_template

constancecchen commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into test/page_template

cee-chen committed:
[APM Cypress tests] harden flaky test

cjcenizal reviewed (approved):
Did a quick smoke test of Management apps and didn't notice anything weird.

cee-chen committed:
[APM Cypress tests] Fix failing Cypress test, again

matschaffer reviewed (approved):
LGTM - I opened https://github.com/elastic/kibana/issues/140149 for @elastic/infra-monitoring-ui to clean up.

Is there a specific migration guide? If so can you point it out? I didn't see anything that looked quite like a step by step from old to new. Would be helpful to have specific instructions I think.

juliaElastic reviewed (approved):
Fleet change LGTM

sphilipse commented:
Enterprise Search lgtm

gbamparop commented on x-pack/plugins/apm/ftr_e2e/cypress/integration/read_only_user/service_overview/service_overview.spec.ts:
```diff
@@ -226,7 +226,9 @@ describe('Service Overview', () => {
       );
 
-      cy.get('[data-test-subj="environmentFilter"]').type('production');
+      cy.get('[data-test-subj="environmentFilter"] input').type('production', {
```
Is this change needed?

constancecchen replied:
The APM Cypress tests failed 4 times in a row without it, which technically I can ask KibanaOps to skip the tests for us, but the failure looked solvable and reasonable so I figured I'd add it in. The test _is_ flaky, but I think this resolves that.

gbamparop replied:
I see, thanks for doing that!

yuliacech reviewed (approved):
Thanks a lot for working on this EUI update @constancecchen! 
Platform Onboarding changes LGTM, tested locally 👍

constancecchen commented:
> LGTM - I opened https://github.com/elastic/kibana/issues/140149 for https://github.com/orgs/elastic/teams/infra-monitoring-ui to clean up.
> Is there a specific migration guide? If so can you point it out? I didn't see anything that looked quite like a step by step from old to new. Would be helpful to have specific instructions I think.

Thanks a million @matschaffer! There's not a step-by-step migration guide for the old `EuiPageTemplate_Deprecated` to `EuiPageTemplate` but there are some helpful guidelines here: https://elastic.github.io/eui/#/templates/page-template/guidelines

For the other granular components, the migration looks like:

- `EuiPageContent_Deprecated / EuiPageContentBody_Deprecated` ->  use **EuiPageSection** instead
- `EuiPageContentHeader_Deprecated / EuiPageContentHeaderSection_Deprecated` -> use **EuiPageHeader** instead
- `EuiPageSideBar_Deprecated` -> use the new EuiPageSidebar

Please feel free to ping me or the EUI team either in Slack or in your issue for assistance, we'd be super happy to help.

gbamparop reviewed (approved):
APM changes LGTM!

cee-chen committed:
Merge branch 'main' into test/page_template

constancecchen commented:
@elastic/kibana-design @elastic/uptime @dasansol92 

As a heads up, since the major affected solutions/plugins have approved this PR at this point, I'll be asking KibanaOps to admin merge this PR by EOD today. Please do take a look / speak up if you see any issues!

dominiqueclarke reviewed (approved):
Uptime changes LGTM

constancecchen commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into test/page_template

jbudz commented:
Merging - the osquery failures seem to be unrelated, caused by elastic-agent setup failures followed by missing index patterns.
